napari/components/dims.py

```diff
@@ -143,29 +194,66 @@ def ndim(self):
 
     @ndim.setter
     def ndim(self, ndim):
-        if self.ndim == ndim:
+        cur_ndim = self.ndim
+        if cur_ndim == ndim:
             return
-        elif self.ndim < ndim:
-            for i in range(self.ndim, ndim):
-                self.set_initial_dims(0, insert=True)
+        elif ndim > cur_ndim:
+            # Range value is (min, max, step) for the entire slider
+            self._range = [(0, 2, 1)] * (ndim - cur_ndim) + self._range
+            # Point is the slider value if in point mode
+            self._point = [0] * (ndim - cur_ndim) + self._point
+            # Interval value is the (min, max) of the slider selction
+            # if in interval mode
+            self._interval = [(0, 1)] * (ndim - cur_ndim) + self._interval
+            self._mode = [DimsMode.POINT] * (ndim - cur_ndim) + self._mode
+            self._order = list(range(ndim - cur_ndim)) + [
+                o + ndim - cur_ndim for o in self.order
+            ]
+            # Default axis labels go from "-ndim" to "-1" so new axes can easily be added
+            self._axis_labels = [
+                str(i - ndim) for i in range(ndim - cur_ndim)
+            ] + self._axis_labels
```

Comment:
I also don't love the negative indices

Reply:
that was me - reason was that new axes add at the beginning not the end, so let's say you add a 2D image you'd have [0, 1], then if you add a 4D image afterward you'd really get [2, 3, 0, 1], whereas if you just added a 4D image it would be [0, 1, 2, 3]. To get around this I went with `-` for the default. Now maybe this just exposes a deeper problem, but adding new axes at the beginning was done to be consistent with broadcasting behavior

Reply:
@sofroniewn why can't you get [0, 1, 2, 3] when you add a 4D image? ie rename [0, 1] to [2, 3], then prepend [0, 1]?

Reply:
do we always rename? what if someone had set `['y', 'x]`. Maybe we should? It's not clear that `['-3', '-2', 'y', 'x']` or `['0', '1', 'y', 'x']` is preferable, so maybe forcing it to `['0', '1', '2', '3']` makes the most sense

Reply:
That I don't know, it's a bit of a corner case. My instinct is to only rename if the existing axes are the default, otherwise, add [0, 1, ..., `however_many_you_need`] to the front. But I don't think it's a big deal, the main Q is what to name prepended axes, and I think [0, 1, ...] makes the most sense, much more than negative values. I'd rather have empty strings than negative values tbh!

Reply:
ok - let's make this here
```python
self._axis_labels = [str(i) for i in range(ndim - cur_ndim)] + self._axis_labels
```
and
```python
self._axis_labels[axis] = str(axis)
```
down below. We'll also need to change the values in the test. I think these are a more sensible starting point, and we can examine the corner-case behavior as it comes up

Reply:
Wait, wouldn't that code result in `[0, 1, 0, 1]` being the labels?

Reply:
yes if someone went from 2D -> 4D it would, i thought we could accept that for now and revisit in the future though as I don't think that is a very common workflow at this point

Reply:
What? It's literally one of my examples in the blog post 😂, make a 2D image then overlay a 3D+ array for parameter scanning. I don't think it's too onerous to use instead:

```
if self._axis_labels == list(map(str, range(cur_ndim))):
    self._axis_labels = list(map(str, range(ndim)))
else:
    self._axis_labels = list(map(str, range(ndim - cur_ndim))) + self._axis_labels
```

Reply:
yes - that looks like a better suggestion!!